integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/CacheTestCase.java

```diff
@@ -0,0 +1,105 @@
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import java.io.File;
+import javax.xml.xpath.XPathExpressionException;
+
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Includes tests for the cache mediator
+ *
+ * Currently tests the maxMessageSize and maxSize parameters in the cache mediator.
+ */
+public class CacheTestCase extends ESBIntegrationTest {
+
+    @BeforeClass(alwaysRun = true)
+    public void deployArtifacts() throws Exception {
+        super.init();
+        loadESBConfigurationFromClasspath(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "mediatorconfig"
+                        + File.separator + "cache" + File.separator + "CacheApis.xml");
+
+    }
+
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    @Test(groups = {"wso2.esb"}, description = "Test maxMessageSize value is smaller than the message size")
+    public void testSmallMessageSize() throws AxisFault, XPathExpressionException, InterruptedException {
+        OMElement response;
+
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "IBM");
+        String firstResponse = response.getFirstElement().toString();
+
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "IBM");
+
+        if (!firstResponse.equalsIgnoreCase(response.getFirstElement().toString())) {
+            assertTrue(true, "The size of messages received is greater than 1000");
+        }
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void close() throws Exception {
+        super.cleanup();
+    }
+
+    @SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+    @Test(groups = {"wso2.esb"}, description = "Tests whether maxSize value evicts the new messages")
+    public void testMaxSize() throws AxisFault, XPathExpressionException, InterruptedException {
+        OMElement response;
+
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "ABCD");
+        String responseABCD1 = response.getFirstElement().toString();
+
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "ABC");
+        String responseABC1 = response.getFirstElement().toString();
+
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "AB");
+        String responseAB1 = response.getFirstElement().toString();
+
+        response = axis2Client.sendSimpleStockQuoteRequest(getApiInvocationURL("maxMessageOrSize"), "", "AB");
+        String responseAB2 = response.getFirstElement().toString();
+
+        if (responseAB1.equalsIgnoreCase(responseAB2)) {
+            assertTrue(true, "The response AB is cached");
```

Comment:
Same as above, shall we correct in all places?

Reply:
Fixed in #983